lib/extensions/attachable/active_record/base.rb

```diff
@@ -16,44 +18,50 @@ def has_many_attachments # rubocop:disable Style/PredicateName
     def has_one_attachment # rubocop:disable Style/PredicateName
       include SingularInstanceMethods
 
-      validates :attachments, length: { maximum: 1 }
+      validates :attachment_references, length: { maximum: 1 }
 
-      has_many :attachments, as: :attachable, class_name: "::#{Attachment.name}",
-                             inverse_of: :attachable, dependent: :destroy, autosave: true
+      has_many :attachment_references, as: :attachable, class_name: "::#{AttachmentReference.name}",
+                                       inverse_of: :attachable, dependent: :destroy, autosave: true
     end
   end
 
   module SingularInstanceMethods
     ATTACHMENT_ATTRIBUTE = 'attachment'.freeze
 
-    def attachment
-      attachments.take
+    def attachment_reference
+      attachment_references.take
     end
+    # AttachmentReference is the duck type of Attachment, so we alias the method here.
```

Comment:
Attachment references can substitute attachments, so allow access using the `attachment` identifier